server/src/main/java/org/apache/druid/metadata/segment/SqlSegmentsMetadataManagerV2.java

```diff
@@ -84,16 +85,21 @@ public SqlSegmentsMetadataManagerV2(
     this.managerConfig = managerConfig.get();
     this.segmentMetadataCache = segmentMetadataCache;
     this.schemaConfig = centralizedDatasourceSchemaConfig;
+
+    // Segment metadata cache currently cannot handle schema updates
+    if (segmentMetadataCache.isEnabled() && schemaConfig.isEnabled()) {
+      throw InvalidInput.exception(
+          "Segment metadata incremental cache and segment schema cache cannot be used together."
```

Comment:
nit: persona for this error message should be `OPERATOR`, but `InvalidInput` uses persona `USER`. I think it doesn't really matter much though, since this error isn't going to go out on an HTTP API, so the special DruidException persona and category stuff won't be used. It could just as well be a regular `IllegalArgumentException` or `UnsupportedOperationException`.

Reply:
another nit: would be nice to include the property names in the error message, so operators know what properties to search their config files for.